python/GafferUI/PathListingWidget.py

```diff
@@ -904,6 +904,21 @@ def event( self, event ) :
 
 		return QtWidgets.QTreeView.event( self, event )
 
+	def resizeEvent( self, event ) :
+
+		QtWidgets.QTreeView.resizeEvent( self, event )
+
+		if self.header().isHidden() :
+			# geometriesChanged is not emitted on resize while the header is hidden,
+			# so we need to call __geometriesChanged ourselves here to ensure
+			# column widths are updated.
+			## \todo It seems unnecessary to perform this update as either part of
+			# resizeEvent or geometriesChanged depending on header visibility.
+			# Relying only on resizeEvent also appears to improve column sizing on
+			# macOS but introduces issues with scrollbars sometimes overlapping
+			# the header on all platforms, which will need to be investigated...
+			self.__geometriesChanged()
```

Comment:
Some of the logic in `__geometriesChanged()` seems a little odd with respect to being called from `resizeEvent()`. At least in my testing the `self.header().length() <= ...` conditional seems to always be `True` in that context. It seems to makes sense that we would always resize the stretch columns when the widget resizes, but that makes me think I probably never really understood the purpose of the conditional in the first place (and/or what is different in the `geometriesChanged` context).

Reply:
I  was noticing the overlap more with the Hierarchy View, where a fair portion of the Exclusions column was being overlapped by the scrollbar. Updating `__resizeStretchColumns`  to use the viewport width, rather than the header width improves this, shrinking the Name column to keep all of the Exclusions column visible beside rather than behind the scrollbar in that situation.

![scrollbarOverlap](https://user-images.githubusercontent.com/50844517/225830686-f6f8d2bc-3bd9-44b0-aa48-d28c6c190900.png)

The conditional is there to prevent the stretch columns from being resized in situations where the user has manually adjusted columns such that their total width is greater than the viewport width. Without it, resizing the widget would result in the stretch columns immediately shrinking in order to fit all columns back within the available width. If the user hasn't adjusted the column widths, then I'd expect the  `self.header().length() <= ...` conditional to generally be `True` like you've observed. I've added a (very eloquently named) `testColumnsWiderThanViewportPreventStretch` test covering this behaviour.